eZ/Publish/Core/Persistence/Cache/CacheServiceDecorator.php

```diff
@@ -101,12 +148,90 @@ private function washKey($key)
      * Clears the cache for the key, or if none is specified clears the entire cache. The key can be either
      * a series of string arguments, or an array.
      *
-     * @internal param array|null|string $key , $key, $key...
+     * @internal
+     * @param array|null|string $key , $key, $key...
+     * @return bool
      */
-    public function clear()
+    public function clear(...$key)
+    {
+        // Make washed string key out of the arguments
+        if (empty($key)) {
+            $key = self::SPI_CACHE_KEY_PREFIX;
+        } elseif (!isset($key[1]) && is_array($key[0])) {
+            $key = self::SPI_CACHE_KEY_PREFIX . '/' . implode('/', array_map([$this, 'washKey'], $key[0]));
+        } else {
+            $key = self::SPI_CACHE_KEY_PREFIX . '/' . implode('/', array_map([$this, 'washKey'], $key));
+        }
+
+        return $this->rawClear([$key]);
+    }
+
+    private function rawClear(array $keys)
```

Comment:
Naming: feels to me there's much more going on than just "raw" clearing. I'd understood if `executeClear` below would be called `rawClear`, but not this one.

My suggestion would be also a bit weird, but at least self-commenting:
```suggestion
    private function clearOrDefer(array $keys)
```